src/utils/common.ts

```diff
@@ -405,3 +405,39 @@ export function denormalizeList<ListItem, SecondaryListItem, ReturnType>(
 
     return newList;
 }
+
+export function hasSomeDefinedValue(item: unknown) {
+    if (isNotDefined(item)) {
+        return false;
+    }
+
+    if (typeof item === 'boolean') {
+        return true;
+    }
+
+    if (typeof item === 'number') {
+        return !Number.isNaN(item);
+    }
+
+    if (typeof item === 'string') {
+        return isTruthyString(item.trim());
+    }
+
+    if (Array.isArray(item)) {
+        if (item.length === 0) {
+            return false;
+        }
+
+        return item.some(hasSomeDefinedValue);
+    }
+
+    if (typeof item === 'object') {
+        if (!item) {
+            return false;
+        }
```

Comment:
Do we even need this check?

Reply:
typescript will cast it as `object | null`

Reply:
We have already used isDefined on the top. So, it shouldn't be the case.